File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        mx = max(candies)
        ans =[]
        for it in candies:
            if(it + extraCandies >= mx): ans.append(True)
            else: ans.append(False)
        return ans

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.go
package main

import "fmt"

func kidsWithCandies(candies []int, extraCandies int) []bool {
	ans := make([]bool, len(candies))
	mx := 0
	for _, candy := range candies {
		mx = max(mx, candy)
	}
	for i, it := range candies {
		if it+extraCandies >= mx {
			ans[i] = true
		}
	}
	return ans
}